src/Entity/PropertyWrapper/BackedEnumWrapper.php

```
@@ -42,9 +45,16 @@ public function convertFromRawValue(mixed $value): ?BackedEnum
 			throw new NullValueException($this->propertyMetadata);
 		}
 
-		assert(is_int($value) || is_string($value));
 		$type = array_key_first($this->propertyMetadata->types);
-		assert(is_subclass_of($type, BackedEnum::class));
-		return $type::from($value);
+		if (is_int($value) || is_string($value)) {
+			if (is_subclass_of($type, BackedEnum::class)) {
+				return $type::from($value);
+			}
+		}
+		if ($value instanceof BackedEnum && $value instanceof $type) {
+			return $value;
+		}
+
+		return null;
```

This seems dangerous to silently fallback to null.

---

irrelevant